steward/src/commands/keys/eth/import.rs

```diff
@@ -38,39 +41,63 @@ impl Runnable for ImportEthKeyCmd {
             }
         }
 
-        let mnemonic = match self.mnemonic.clone() {
-            Some(mnemonic) => mnemonic,
-            None => rpassword::read_password_from_tty(Some("> Enter your bip39 mnemonic:\n"))
-                .expect("Could not read mnemonic"),
-        };
-
-        let key = match bip32::Mnemonic::new(mnemonic.trim(), Default::default()) {
-            Ok(mnemonic) => {
-                let seed = mnemonic.to_seed("");
-
-                let path = config.ethereum.key_derivation_path.trim();
-                let path = path
-                    .parse::<bip32::DerivationPath>()
-                    .expect("Could not parse derivation path");
-
-                let key = bip32::XPrv::derive_from_path(seed, &path).expect("Could not derive key");
-                let key = k256::SecretKey::from(key.private_key());
-                key.to_pkcs8_der()
-                    .expect("Could not PKCS8 encod private key")
-            }
-            Err(_) => {
-                let key = rpassword::read_password_from_tty(Some("> Enter your private-key:\n"))
-                    .expect("Could not read private-key");
+        if self.key.is_some() {
+            let key = match self.key.clone() {
+                Some(private_key) => private_key,
+                None => rpassword::read_password_from_tty(Some("> Enter your private-key:\n"))
```

Comment:
This condition will never be hit, because you're checking `is_some` before ever running the `match`. We would either need two different import commands, or some kind of configuration inside this command, to handle both command-line provided keys or keys obtained over a TTY.

Reply:
I think the best approach is to implement two import commands